src/net/sourceforge/kolmafia/StaticEntity.java

```diff
@@ -81,17 +81,29 @@ public static final String getVersion() {
     if (StaticEntity.cachedVersionName == null) {
       StringBuilder versionName =
           new StringBuilder(PRODUCT_NAME).append(" r").append(StaticEntity.getRevision());
+      if (isCodeModified()) {
+        versionName.append("-M");
+      }
       StaticEntity.cachedVersionName = versionName.toString();
     }
-
     return StaticEntity.cachedVersionName;
   }
 
+  private static boolean isCodeModified() {
+    Attributes attributes = getAttributes();
+    if (attributes != null) {
+      String buildRevision = attributes.getValue("Build-Dirty");
+      return buildRevision.equals("true");
+    }
+    return false;
+  }
+
```

Comment:
This is obviously completely optional but I would always write this like:

```suggestion
  private static boolean isCodeModified() {
    Attributes attributes = getAttributes();
    if (attributes == null) {
      return false;
    }
  
    return attributes.getValue("Build-Dirty").equals("true");
  }
```

in the spirit of return early and indent less